test/spec/services/homePagePreferenceSpec.js

```diff
@@ -29,39 +29,57 @@ describe("HomePagePreferenceService", function(){
 
    });
 
-  describe("#getHomePagePath", function(){
-    it("default set to catalog-home",function(){
-      var homePagePref = {};
-      HomePagePreferenceService.setHomePagePreference(homePagePref);
-      var homePagePreference = HomePagePreferenceService.getHomePagePath();
-      expect(homePagePreference).toEqual("/catalog");
-    });
+   describe("#getHomePagePath", function(){
+     describe('when given an empty config object', function() {
+       it("should set the default home page to /catalog",function(){
+         var homePagePref = {};
+         HomePagePreferenceService.setHomePagePreference(homePagePref);
+         var homePagePreference = HomePagePreferenceService.getHomePagePath();
 
-    it("set project list", function(){
-      var homePagePref = {"type":"project-list"};
-      HomePagePreferenceService.setHomePagePreference(homePagePref);
-      var homePagePreference = HomePagePreferenceService.getHomePagePath();
-      expect(homePagePreference).toEqual("/projects");
-    });
+         expect(homePagePreference).toEqual("/catalog");
+       });
+     });
 
-    it("set to project overview",function(){
-      var homePagePref = {
-          "type":"project-overview",
-          "project":"test-pro"
-      };
-      HomePagePreferenceService.setHomePagePreference(homePagePref);
-      var homePagePreference = HomePagePreferenceService.getHomePagePath();
-      expect(homePagePreference).toEqual("/project/test-pro/overview?isHomePage=true");
-    });
+     describe('when given a config object with `type` set to `project-list`', function() {
+       it("should set the home page to /projects", function(){
+         var homePagePref = {"type":"project-list"};
+         HomePagePreferenceService.setHomePagePreference(homePagePref);
+         var homePagePreference = HomePagePreferenceService.getHomePagePath();
+
+         expect(homePagePreference).toEqual("/projects");
+       });
+     });
+
+     describe('when given a config object with `type` set to `project-overview`', function() {
+       it("should set the home page to the overview of the provided project at /project/test-pro/overview",function(){
+         var homePagePref = {
+             "type":"project-overview",
+             "project":"test-pro"
+         };
+         HomePagePreferenceService.setHomePagePreference(homePagePref);
+         var homePagePreference = HomePagePreferenceService.getHomePagePath();
+
+         expect(homePagePreference).toEqual("/project/test-pro/overview?isHomePage=true");
+       });
+     });
+
+     describe('when given a config object with an invalid `type` option', function() {
+       // in the UI, this will also trigger a notification that the project is
+       // invalid.  This additional functionality should be verified either with
+       // a unit test or an e2e test against the appropriate controller.
+       it("should not set the home page", function(){
```

Comment:
This is really testing that `notificationInvalidProjectHomePage` clears the value. The description is better, but not quite accurate.

Reply:
True, this should be in a separate `describe('#notifyInvalidProjectHomePage')`.  I'll update.

Reply:
Updated.